File: backend/DataAccess/Entities/Destination.cs
using System.Diagnostics;

namespace DataAccess.Entities
{
    public class Destination
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public DateTime ArrivalDate { get; set; }
        public DateTime DepartureDate { get; set; }
        public string? Notes { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public int TripId { get; set; }
        
        public Trip Trip { get; set; } = null!;
        public ICollection<Activity> Activities { get; set; } = new List<Activity>();
    }
}

File: frontend/src/models/Destination.ts
export interface Destination {
  id: number;
  name: string;
  location: string;
  arrivalDate: string;
  departureDate: string;
  notes?: string;
  latitude?: number;
  longitude?: number;
  tripId: number;
}

export interface CreateDestinationRequest {
  name: string;
  location: string;
  arrivalDate: string;
  departureDate: string;
  notes?: string;
  latitude?: number;
  longitude?: number;
}

export interface UpdateDestinationRequest {
  name: string;
  location: string;
  arrivalDate: string;
  departureDate: string;
  notes?: string;
  latitude?: number;
  longitude?: number;
}